### PR TITLE
[luci] Support quantization of ResizeBilinear

### PR DESCRIPTION
**Pull request by dayo09**
This commit supports quantization and its unittest of ResizeBilinear operation.

ONE-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

------------

For #6367

### CODE REVIEW

**Comment by dayo09** [comment]
```
C++ exception with description "loco::must_cast() failed to cast: PN4luci10CircleNodeE" thrown in the test body.
```
At TEST_WITH_GRAPH...

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 1]
```diff
@@ -803,6 +829,30 @@ class PowTestGraph final : public SimpleTestGraph
...
+{
+public:
+  void init(void)
+  {
+    TestIOGraph::init({1, 4, 4, 1}, {1, 8, 8, 1});
+
+    _size = create_const<Type::S32, int32_t>(g(), {2}, {8, 8});// TODO : check
+    _resizebilinear = g()->nodes()->create<luci::CircleResizeBilinear>();
+    {
+      _resizebilinear->input(input());
```
Comment (jinevening) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:842]: Maybe size is missing?

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:844]: @jinevening 
It still throws the same exception 😢

**Comment by dayo09** [comment]
BTW it's so inconvinient to find out the exception information.

```
TEST(QuantizedModelVerifierTest, ResizeBilinear)
{
  {
    // expand test to know the exception
    ResizeBilinearTestGraph g;
    g.init();
    quantize_and_verify(g.g(), Type::U8, Granularity::LayerWise);
    
    luci::QuantizeWithMinMaxPass pass(Type::FLOAT32, Type::U8, Granularity::LayerWise);
    pass.run(g.g());

    luci::QuantizedModelVerifier verifier(Type::U8, Granularity::LayerWise);
    verifier.verify(g.g());
  }

  TEST_WITH_GRAPH(ResizeBilinearTestGraph, Type::U8, Granularity::LayerWise);
  TEST_WITH_GRAPH(ResizeBilinearTestGraph, Type::U8, Granularity::ChannelWise);
  TEST_WITH_GRAPH(ResizeBilinearTestGraph, Type::S16, Granularity::ChannelWise);
  SUCCEED();
}
```

**Comment by dayo09** [comment]
### GDB Debugging

The first output node fails at must_cast.

```
luci::QuantizeWithMinMaxPass::run (this=0x7fffffffd8f0, g=0x555555aa1e20)
    at /nfs/git/ONE/compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:1078
1078	{
(gdb) n
1079	  LOGGER(l);
(gdb) n
1080	  INFO(l) << "QuantizeWithMinMaxPass Start" << std::endl;
(gdb) n
1083	  for (auto node : loco::active_nodes(loco::output_nodes(g)))
(gdb) n
1085	    QuantizeActivation qa(_input_dtype, _output_dtype);
(gdb) n
1086	    auto circle_node = loco::must_cast<luci::CircleNode *>(node);
(gdb) n
1087	    circle_node->accept(&qa);
(gdb) n
unknown file: Failure
C++ exception with description "loco::must_cast() failed to cast: PN4luci10CircleNodeE" thrown in the test body.
```

**Comment by dayo09** [comment]
```
// Avoid overriding TestIOGraph::input()
loco::Node *input_rbl() { return _resizebilinear->input(); }
```

Solved: _resizebilinear->input() getter was overriding the graph's input getter. I renamed the function.
@seanshpark Thanks for the help!! 😄

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 2]
```diff
@@ -828,6 +828,32 @@ class PowTestGraph final : public SimpleTestGraph
...
+      _resizebilinear->size(_size);
+    }
+    output()->from(_resizebilinear);
+
+    set_minmax_to_non_const(g(), -1, 1);
+  }
+
+  // Avoid overriding TestIOGraph::input()
+  loco::Node *input_rbl() { return _resizebilinear->input(); }
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:850]: I cann't see where this is used. Will it be used in another PR?

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:850]: @seanshpark Sorry, it's not yet rebased.. I cannot fetch from the upstream for now

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:850]: Oh, the answer to what you point out is: no, it isn't used. I will remove it. 😄

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 3]
```diff
@@ -828,6 +828,28 @@ class PowTestGraph final : public SimpleTestGraph
...
+    {
+      _resize_bilinear->input(input());
+      _resize_bilinear->size(_size);
+    }
+    output()->from(_resize_bilinear);
+
+    set_minmax_to_non_const(g(), -1, 1);
+  }
+
+  luci::CircleResizeBilinear *_resize_bilinear = nullptr;
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:849]: it would be better to add scope `private:` above this line. Other `XXXTestGraph` classes uses `public:` but I think it would be better to change them also.

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:849]: @seanshpark Sure :-D

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:849]: > but I think it would be better to change them also(in another PR)

I will make a change in a separate PR. :-D

Comment (seanshpark) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:849]: There is no `private:` scope here. Please add in this PR for this file.

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:850]: Sure. updated

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Comment by seanshpark** [comment]
Merge as @jinevening is on vacation this week :)
